template_test.go

```diff
@@ -30,6 +30,7 @@ func TestTemplate(t *testing.T) {
 
 	// Step 1: Create a brand new template
 	id, err := client.CreateTemplate(ctx, template)
+	defer client.DeleteTemplate(ctx, id)
```

Comment:
This call is not necessary. The template is deleted later in this test: https://github.com/transloadit/go-sdk/pull/34/files#diff-abbcc27f0a33a29d477f738337cf84198c320e4847cb9a278dc3e892f167ceb8R100 In fact, I assume this delete call here will always error out, but the error is never checked.

Reply:
I agree with you that the defer will failed in nominal case but if there is a failure in the test, the t.Error / t.Fatal leaves the function and the template is not deleted . With defer, I have my "finally" block that cleans the resource.

Reply:
Does it happen regularly for you that this test case fails and leaves the template in your account?

Reply:
No but it is cleaner

Reply:
Personally, I prefer if failed test cases leave the resources untouched, so they can be inspected to find the root cause. But since this is not happening often enough, I think we are fine here.